tests/integration/test_authd/test_api_agent_registration.py

```diff
@@ -0,0 +1,249 @@
+            client_key_agent_list = entry.split()
+
+            client_key_agent_dict = {}
+            client_key_agent_dict.update({'id': client_key_agent_list[0], 'name': client_key_agent_list[1],
+                                          'ip': client_key_agent_list[2], 'key': client_key_agent_list[3]})
+
+            client_keys_dictionary.append(client_key_agent_dict)
+
+    desired_entries = []
+    for client_keys_entry_dict in client_keys_dictionary:
+        if agent_parameters.items() <= client_keys_entry_dict.items():
+            desired_entries.append(agent_parameters)
+    return desired_entries
+
+
+@pytest.fixture(scope="module")
+def clean_registered_agents():
+    control_service('stop')
+    truncate_file(CLIENT_KEYS_PATH)
+    control_service('start')
+
+    yield
+
+    control_service('stop')
+    truncate_file(CLIENT_KEYS_PATH)
+    control_service('start')
+
+
+@pytest.fixture(scope="module")
+def truncate_api_log():
+    truncate_file(API_LOG_FILE_PATH)
+
+
+def check_valid_agent_id(id):
+    return re.match("(^[0-9][0-9][0-9]$)", id)
+
+
+def check_valid_agent_key(key):
+    return len(key) > 0
+
+
+def check_api_data_response(api_response, expected_response):
+    api_response_error = api_response['error']
+    assert api_response_error == expected_response['error'], f"Expected API response \
+                                                               {expected_response} but {api_response_error} \
+                                                               was received instead"
+
+    if api_response_error == 0:
+        if 'key' in expected_response['data']:
+            assert check_valid_agent_key(api_response['data']['key']), f"Invalid agent key received:\
+                                                                        {api_response['data']['key']}"
+            del api_response['data']['key']
+            del expected_response['data']['key']
+
+        if 'id' in expected_response['data']:
+            assert check_valid_agent_id(api_response['data']['id']), f"Invalid id received:\
+                                                                        {api_response['data']['id']}"
+            del api_response['data']['id']
+            del expected_response['data']['id']
+
+    return api_response == expected_response
+
+
+@pytest.mark.filterwarnings('ignore::urllib3.exceptions.InsecureRequestWarning')
+@pytest.mark.parametrize("api_registration_parameters", api_registration_requests, ids=api_registration_requests_ids)
+def test_agentd_server_configuration(get_configuration, configure_environment, truncate_api_log,
+                                     clean_registered_agents, wait_for_start_module, api_registration_parameters):
+    '''
+    description:
+        Checks `wazuh-api` responds correctly to agent registration requests. Also, ensure client.keys is update
+        accordingly to the new agents parameters.
+
+    wazuh_min_version:
+        4.4.0
+
+    parameters:
```

Comment:
We need to add `truncate_api_log` and `wait_for_start_module`. Also, `restart_and_wait_api` is not used.

Reply:
Done in 
d55cbfb